### DISCUSSION TITLE
Upgrade path from 2.8.2 to 3.2.0

### DISCUSSION THREAD

**Discussion by jsrolon** [cobbler/cobbler#3705] [Q&A]
I'm dealing with an inherited legacy system here. Due to policy, I'm also stuck with Centos 7. My understanding is that this OS only supports up to cobbler 3.2.0.

My current cobbler is v2.8.2. I built 3.2.0 from source and got an rpm that I then proceeded to install. I see that it created a couple of new files like `/etc/cobbler/settings.rpmnew` which contains a bunch of keys that my previous settings file doesn't contain.

I don't know if keys were removed/changed and how to get the data from 2.8.2 imported to 3.2.0. I have a few thousand systems, and ~10 profiles and distros. `cobbler check` doesn't seem to be too helpful verifying whether everything is ok.

Is there an upgrade guide from 2.8.2 to 3.2.0? I'm wary of forgetting to modify some file somewhere and losing data.

**Comment by SchoolGuy** [comment]
So you do want to do this in two steps:

- Upgrade your instance to 2.8.5 and check that everything works.
- Upgrade to the latest 3.2.x release and then check again.

I saw that sadly we don't have CentOS 7 in our repositories for 3.2.x and as such I cannot offer you the community-maintained RPMs. The reason why the upgrade from 2.8.x to 3.x.x is so bumpy is because the repository changed maintainers during that time and as such much knowledge was lost sadly.

However, the most important things should be automatically scripted and as such only a few things should require your attention. Please make backups as downgrades are not taken into account in the migration paths.

**Reply by jsrolon** [reply]
Thanks for the quick reply!

> However, the most important things should be automatically scripted and as such only a few things should require your attention.

Could you please expand a bit on this? When you say that they're automatically scripted, does it mean that scripts will run during `yum install` of the RPM, or do I have to run scripts myself? If the latter, which scripts should I run?

Also, the most important thing is the last step. What are the few things that will require my attention?

**Reply by SchoolGuy** [reply]
You have two scripts that are of interest (assuming you go to 3.2.2):

- `[migrate-data-v2-to-v3.py](https://github.com/cobbler/cobbler/blob/v3.2.2/scripts/migrate-data-v2-to-v3.py)`
- `[settings-migration-v1-to-v2.sh](https://github.com/cobbler/cobbler/blob/v3.2.2/scripts/settings-migration-v1-to-v2.sh)`

The docs for 3.2.2 can be found here: https://cobbler.readthedocs.io/en/v3.2.2/

The manual migration from 2.8.5 I never attempted but so far all users said that their scripting around Cobbler got messed up. Sadly as such I don't have more insight.

**Reply by djtbs1** [reply]
I have similar issue. I'm moving to RH8.10. On RHEL 7.x using Cobbler 2.8.5. From a known, working installation fully scripted to a broken one on 3.x because parameter options changed or no longer work. For example adding a profile used to be like this:
        cobbler profile add --name=${DISTRO_NAME}-aserver-ng2 \
            --kickstart=/var/lib/cobbler/kickstarts/aserver_rh${PVN}_ng2.ks --kopts="${KOPTS}"
            
BUT "--kopts" is no longer a valid parameter, it changed to the fully written "--kernel-options".
That's not such a big deal, but also "--kickstart" is no longer a valid option and it's not clear what I do here.... the documents aren't clear how you "migrate" this. But I am going to start a new thread / question for this....
